solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseCloudSolrClient.java

```diff
@@ -994,12 +991,15 @@ public RouteException(ErrorCode errorCode, NamedList<Throwable> throwables, Map<
           // and we could not get any information from the server
           //it is probably not worth trying again and again because
           // the state would not have been updated
-          log.info("trying request again");
+          log.info("Request to collection {} failed due to ({}) {}, retry={} commError={} errorCode={} - retrying",
+              inputCollections, errorCode, rootCause, retryCount, wasCommError, errorCode);
           return requestWithRetryOnStaleState(request, retryCount + 1, inputCollections);
         }
       } else {
         log.info("request was not communication error it seems");
       }
+      log.error("Request to collection {} failed due to ({}) {}, retry={} commError={} errorCode={} ",
```

Comment:
Since you add ` - retrying` to the info logging above, maybe add something like ` - max retries hit` here

Reply:
`max retries hit` is a little tricky to determine exactly since e.g. `commError` and other things influence the logic. But inclusion of `MAX_STALE_RETRIES` in the logging could make it clear(er) when "max retries hit" was likely the case.